kafkautils/topic.go

```diff
@@ -32,71 +36,41 @@ const (
 )
 
 type Topic struct {
-	Env           env
-	MsgType       msgType
-	Author        string
-	ConnectorName string
-	Version       *semver.Version
-	EventName     string
-	pb            proto.Message // create an empty protobuf struct instance, filled upon UnmarshalProto
+	Env     env
+	MsgType msgType
+	Schema  string
+	Version string        // can parse into int when necessary
+	pb      proto.Message // create an empty protobuf struct instance, filled upon UnmarshalProto
 }
 
-// String generates the topic string
 func (t Topic) String() string {
-	return strings.Join([]string{string(t.Env), string(t.MsgType), t.Schema()}, TopicDelimiter)
-}
-
-// Schema generates the schema string
-func (t Topic) Schema() string {
-	return strings.Join([]string{
-		t.Author,
-		t.ConnectorName,
-		strings.ReplaceAll(t.Version.String(), ".", "_"),
-		t.EventName,
-	}, TopicDelimiter)
-}
-
-func NewTopic(en, ty, author, connectorName string, version *semver.Version, msg proto.Message) Topic {
-	return Topic{
-		Env:           env(en),
-		MsgType:       msgType(ty),
-		Author:        author,
-		ConnectorName: connectorName,
-		Version:       version,
-		EventName:     strings.ReplaceAll(strings.ToLower(string(msg.ProtoReflect().Descriptor().FullName())), ".", "_"),
```

Comment:
Did you intend to remove this? This generates the eventname directly from the proto.Message

Reply:
no 😅